### PR TITLE
Add GitHub Workflow Replacement for Jenkins job_PreCommit_Kotlin_Examples

### PR DESCRIPTION
**Pull request by aleksandr-dudko**
The following jenkins jobs have been migrated:

https://ci-beam.apache.org/view/PreCommit/job/beam_PreCommit_Kotlin_Examples_Commit/
https://ci-beam.apache.org/view/PreCommit/job/beam_PreCommit_Kotlin_Examples_Cron/
https://ci-beam.apache.org/view/PreCommit/job/beam_PreCommit_Kotlin_Examples_Phrase/

How has it been tested?
The job has been tested on Self-hosted GKE runners ubuntu 20.04.
The following test cases are passed:

by creating pull requests with changes in trigger paths
by adding a comment to the PR
by making a push

image

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by aleksandr-dudko** [comment]
assign to next reviewer: @andreydevyatkin

File: .github/workflows/job_PreCommit_Kotlin_Examples.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: PreCommit Kotlin Examples
+
+on:
+  push:
+    tags: ['v*']
+    branches: ['master', 'release-*']
+    paths: ['model/**', 'sdks/java/**', 'runners/flink/**', 'runners/spark/**', 'runners/direct-java/**', 'examples/kotlin/**', 'release/**']
```
Comment (volatilemolotov) [inline .github/workflows/job_PreCommit_Kotlin_Examples.yml:22]: Add the path to workflow file

Comment (aleksandr-dudko) [reply .github/workflows/job_PreCommit_Kotlin_Examples.yml]: It's done.

File: .github/workflows/job_PreCommit_Kotlin_Examples.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: PreCommit Kotlin Examples
+
+on:
+  push:
+    tags: ['v*']
+    branches: ['master', 'release-*']
+    paths: ['model/**', 'sdks/java/**', 'runners/flink/**', 'runners/spark/**', 'runners/direct-java/**', 'examples/kotlin/**', 'release/**', '.github/workflows/job_PreCommit_Kotlin_Examples.yml']
```
Comment (andreydevyatkin) [inline .github/workflows/job_PreCommit_Kotlin_Examples.yml:22]: Сould you list the paths on each line to make it more readable? Here is an example - https://github.com/apache/beam/blob/master/.github/workflows/beam_PreCommit_Java_Examples_Dataflow.yml#L25-L31

Comment (aleksandr-dudko) [reply .github/workflows/job_PreCommit_Kotlin_Examples.yml]: It's done.

File: .github/workflows/job_PreCommit_Kotlin_Examples.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+jobs:
+  KotlinExamplesPreCommit:
```
Comment (andreydevyatkin) [inline .github/workflows/job_PreCommit_Kotlin_Examples.yml:62]: We should use the same job name as Jenkins to keep it consistent. Please rename to `job_PreCommit_Kotlin_Examples`

File: .github/workflows/job_PreCommit_Kotlin_Examples.yml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    name: Kotlin Examples
```
Comment (andreydevyatkin) [inline .github/workflows/job_PreCommit_Kotlin_Examples.yml:63]: Rename to `job_PreCommit_Kotlin_Examples`

**Review by andreydevyatkin — COMMENTED** [review]
see the inline comments

File: .github/workflows/beam_PreCommit_Kotlin_Examples.yml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  deployments: read
+  id-token: none
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
```
Comment (andreydevyatkin) [inline .github/workflows/beam_PreCommit_Kotlin_Examples.yml:60]: Please add the `concurrency` section. Here is an example: https://github.com/apache/beam/blob/master/.github/workflows/beam_PreCommit_Java_Examples_Dataflow.yml#L47-L49

File: .github/workflows/beam_PreCommit_Kotlin_Examples.yml [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        uses: actions/setup-java@v3.8.0
+        with:
+          distribution: 'zulu'
+          java-version: '8'
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          cache-read-only: false
+      - name: run Kotlin Examples script
+        run: ./gradlew :examples:kotlin:preCommit
```
Comment (andreydevyatkin) [inline .github/workflows/beam_PreCommit_Kotlin_Examples.yml:94]: We should use the gradle action to run the command in order to add the required params - https://github.com/apache/beam/blob/master/.github/actions/gradle-command-self-hosted-action/action.yml

**Review by andreydevyatkin — CHANGES_REQUESTED** [review]
see the comments

**Review by andreydevyatkin — APPROVED** [review]
LGTM

**Review by damccorm — APPROVED** [review]
Thanks!

**Pull request merged**